File: 백준/Silver/2644. 촌수계산/촌수계산.cc
#include <iostream>
using namespace std;

int N, M, A, B, X, Y;
short graph[101][101];
short visited[101];
int queue[101];

int main()
{
    ios::sync_with_stdio(0);
    cin.tie(0);
    cout.tie(NULL);

    cin >> N;
    cin >> A >> B;
    cin >> M;

    for (int i = 0; i < M; i++)
    {
        cin >> X >> Y;

        graph[X][Y] = 1;
        graph[Y][X] = 1;
    }

    int front = 0;
    int rear = 0;

    queue[rear++] = A;
    visited[A] = 1;

    while (front < rear) {
        int vertex = queue[front++];

        for (int i = 1; i < N + 1; i++)
        {
            if (graph[vertex][i] == 1 && visited[i] == 0) {
                visited[i] = visited[vertex] + 1;
                queue[rear++] = i;
            }
        }
    }

    cout << visited[B] - visited[A];

    return 0;
}

File: Cpp_Algo/백준/Silver/2644. 촌수계산/촌수계산.py
N = int(input())
a, b = map(int, input().split())
M = int(input())

graph = [[] for _ in range(N + 1)]

for _ in range(M):
    x, y = map(int, input().split())

    graph[x].append(y)
    graph[y].append(x)

answer = 0
visited = [0] * (N + 1)

queue = []
queue.append(a)
visited[a] = 1

while queue:
    vertex = queue.pop(0)

    for next_vertex in graph[vertex]:
        if visited[next_vertex] == 0:
            visited[next_vertex] = visited[vertex] + 1
            queue.append(next_vertex)

print(visited[b] - visited[a])